hazelcast-wm/src/test/java/com/hazelcast/wm/test/WebFilterClientFailOverTests.java

```diff
@@ -0,0 +1,97 @@
+package com.hazelcast.wm.test;
+
+import com.hazelcast.config.FileSystemXmlConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.IMap;
+import org.apache.http.client.CookieStore;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.junit.Test;
+
+import java.io.File;
+
+import static org.junit.Assert.assertEquals;
+
+public abstract class WebFilterClientFailOverTests extends AbstractWebFilterTest {
+
+    String testName;
+    protected WebFilterClientFailOverTests(String serverXml1) {
+        super(serverXml1);
+    }
+
+    protected WebFilterClientFailOverTests(String serverXml1, String serverXml2) {
+        super(serverXml1, serverXml2);
+        testName = serverXml1;
+    }
+
+    @Test
+    public void whenClusterIsDown_enabledDeferredWrite() throws Exception {
+        CookieStore cookieStore = new BasicCookieStore();
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        hz.shutdown();
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+        assertEquals("true", executeRequest("remove", serverPort1, cookieStore));
+        assertEquals("null", executeRequest("read", serverPort1, cookieStore));
+        hz = Hazelcast.newHazelcastInstance(
+                new FileSystemXmlConfig(new File(sourceDir + "/WEB-INF/", "hazelcast.xml")));
+        Thread.sleep(8000);
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
+        assertEquals("value", executeRequest("read", serverPort1, cookieStore));
+    }
+
+    @Test
+    public void whenClusterIsDownAtBeginning_enabledDeferredWrite() throws Exception {
+        hz.shutdown();
+        CookieStore cookieStore = new BasicCookieStore();
+        assertEquals("true", executeRequest("write", serverPort1, cookieStore));
```

Comment:
I would create an assert method:

assertRequest("true", "write", serverPort1, cookieStore)

And cookie stored core be created in the setup so doesn't need to be passed as argument.